### DISCUSSION TITLE
Using `NavigationLinkStore` with TCA 1.0

### DISCUSSION THREAD

**Discussion by appfrosch** [pointfreeco/swift-composable-architecture#2364] [Q&A] [answered]
I am trying to find the best way for me to do navigation in my TCA 1.0 project.

I stumbled over `NavigationLinkStore.init(_:state:action:onTap:destination:label:)` just to realize that it was deprecated.
I was wondering what the best way forward would be–to use a `NavigationStack` in all circustances?

**Comment by mbrandonw** [comment, marked as answer]
Hi @appfrosch, `NavigationLinkStore` has the same deprecation level as `NavigationLink(isActive:)` because it uses `NavigationLink(isActive:)` under the hood. That is how it has been since its first introduction and was not changed in 1.0.

The guidance for using `NavigationLink` in a TCA app is the same as in a vanilla app. If you can target iOS 16 then you are encouraged to use the newer navigation tools, such as `NavigationStack`. If you don't need to use iOS 16, then targeting iOS 15 or lower will get rid of the deprecation warning.

**Reply by appfrosch** [reply]
Thanks a lot-also: this makes a lot of sense!
I appreciate you taking the time for this informative answer 🙌
